crates/swc_fast_ts_strip/tests/errors/modules.ts

```diff
@@ -0,0 +1 @@
+module 'foo' { }
```

Comment:
### Ambient Module Declaration

This is an Ambient Module Declaration, not a `module`-keyword namespace.  It should be stripped as a type.  It should not error.

```ts
module 'myAmbientModuleDeclaration' { }   // this line disappears
```

### `module`-keyword namespace

The use of the legacy [`module`-keyword namespace](https://github.com/microsoft/TypeScript/issues/57913) should error, just like it does for `namespace`-keyword namespaces.

```ts
module aModuleKeywordNamespace { }  // this line errors
```

### type-only namespaces

The use of type-only namespaces should be permitted and stripped like any other type.

`declare namespace Foo { }    // this line disappears`

### type-only module-keyword namespaces

The use of type-only module-keyword namespaces is a grey/opinionated area and depends on whether you want to promote modern code or retain compatibility with legacy usage.  Personally I'd suggest erroring is healthiest because the fix is trivial - the user can replace `module` with `namespace`.

`declare module Foo { }    // probably this line errors`

Reply:
I resolved first three items, and I'll keep the fourth one as-is (for now)